src/utilities/params.ts

```diff
@@ -102,12 +102,8 @@ export function setParamValue(value: unknown, param: Param): string {
     return numberParam.set(value, extras)
   }
 
-  if (isParamGetter(param)) {
-    const stringValue = (value as any).toString()
-
-    param(stringValue, extras)
-
-    return stringValue
+  if (isParamSetter(param)) {
+    return param(value, extras)
```

Comment:
This change doesn't seem right to me. A param cannot be type `ParamSetter`. Only `ParamGetter`. So this check will return true (because both types have the same signature) but then most likely error because its being type guarded as a `ParamSetter` when its actually a `ParamGetter` and a `ParamGetter` expects a string as its value.

Reply:
I see that `Param` cannot be `ParamSetter`, that makes sense. However getting a `ParamGetter` doesn't make sense for SETTING a param value, the contract is backwards. 

I felt super confident in that change when how we had to contort the value to call a getter.  But yeah, I'm not sure how ParamSetter would ever be used by a developer except within a ParamGetSet. 

If a developer provides ONLY a ParamGetter, I don't think we can assume it gives us anything useful for converting back into URL strings. By providing only a getter my expectation would be that all we do on the way back is call String constructor?

Reply:
If only a ParamGetter is provided by the user that means that whatever value they provide for the param when setting can be directly converted to a string. It goes straight into the url. 

But just because it can be directly converted to a string does not mean that string value will be valid when passed into the ParamGetter the user provided. 

So the question is, if a value is given is invalid based on the ParamGetter. When do we error? We either

1. Convert the provided value into a string, create a new url, and then try to match the url with a route. A valid string will match the requested route, an invalid string will not. So we get a NotFound rejection

2. Convert the provided value into a string. Check if it passes the getter and if it does go ahead and create the new url. If it does not error right away. URL does not get updated and no rejection happens. It's just a JavaScript runtime error. 

The second is what I had. Not because I think that's necessarily better (I think creating the url and rejecting is a good experience) but because it's consistent with other get/set patterns. A Boolean for example will error in the setter. And user land GetSets can also error in the setter. 

It also felt weird that if you had a ParamGetter than the unit tests wouldn't error the same way for the get vs the set. 

Totally open to a different implementation for handling sets of ParamGetter.

Reply:
Apologies, maybe I'm being dense here I am behind on sleep lol

You're losing me at "...if a value is given is invalid based on the ParamGetter...".  We already know the route, and the "values" given here refers to the values provided by the developer at runtime calling a routeMethod that ultimately need to be converted to strings so we can form a complete URL. So "values" will be whatever `T` is (boolean, number, etc) NOT the string value. 

If my route is 
```ts
{
  name: 'foo', 
  path: path('/foo/:bar', { bar: Boolean })
  ...
}
```

We're working through what happens when the developer later calls `router.routes.foo(true)`

"values" is `[true]` and needs to be converted to a string to form the URL "foo/true"

"ParamGetters" are responsible for converting a string value into `T`, which I'm not seeing how is supposed to help us here.

It seems like maybe you're confusing assembling a URL given a route and values (with setParamValue) with matching a URL given a url string (with getParamValue)?